lib/galaxy/tools/toolbox/base.py

```diff
@@ -704,6 +704,7 @@ def __watch_directory( self, directory, elems, integrated_elems, load_panel_dict
         def quick_load( tool_file, async=True ):
             try:
                 tool = self.load_tool( tool_file )
+                tool._lineage = self._lineage_map.register( tool )
```

Comment:
Maybe this can be moved inside the `__add_tool()` method (called on the next line and similarly at lines 569-572)?